synapse/storage/databases/main/devices.py

```diff
@@ -1573,11 +1583,11 @@ def _add_device_outbound_poke_to_stream_txn(
         self,
         txn: LoggingTransaction,
         user_id: str,
-        device_ids: Collection[str],
-        hosts: List[str],
+        device_ids: Iterable[str],
+        hosts: Iterable[str],
```

Comment:
I think this has to be a `Collection` as we iterate over it multiple times?

Reply:
Oh, good catch.